src/main/kotlin/pl/droidsonroids/gradle/jacoco/testkit/JacocoTestKitExtension.kt

```diff
@@ -4,31 +4,38 @@ import org.gradle.api.Project
 import org.gradle.api.Task
 import org.gradle.api.provider.Provider
 import org.gradle.api.tasks.TaskProvider
+import org.gradle.api.tasks.WriteProperties
 import org.gradle.testing.jacoco.plugins.JacocoTaskExtension
-import java.io.File
 
 open class JacocoTestKitExtension(private val project: Project) {
-    val jacocoRuntimePathProvider: Provider<String> = project.provider { project.configurations.getByName(Configurations.jacocoRuntime).asPath }
+
+    val jacocoRuntimePathProvider: Provider<String> =
+        project.provider { project.configurations.getByName(Configurations.jacocoRuntime).asPath }
 
     fun applyTo(configurationRuntime: String, taskProvider: TaskProvider<Task>) {
         with(project) {
             val jacocoTestKitPropertiesTask = tasks.register(
                 generatePropertiesTaskName(taskProvider.name),
-                GenerateJaCoCoTestKitProperties::class.java
+                WriteProperties::class.java,
             ) {
-                it.outputFile = File(buildDir, "testkit/${taskProvider.name}/testkit-gradle.properties")
-                it.destinationFile.set(
```

Comment:
I haven't understood what's the difference between destination and output file, since they point at the same location. I followed my intuition and the JacocoTaskExtension's destinationFile is now used only as `org.gradle.jvmargs` property value